include/deal.II/matrix_free/fe_evaluation_data.h

```diff
@@ -121,8 +121,8 @@ class FEEvaluationData
 public:
   static constexpr unsigned int dimension = dim;
 
-  using NumberType = Number;
-
+  using NumberType             = Number;
+  using shape_info_number_type = Number;
```

Comment:
Why not extract this info from `ShapeInfoType`?

Reply:
I plan to do so in a future PR, but I wanted to not mix in changes in `ShapeInfo`. If you prefer, we can also make that change already now.

Reply:
Fine for now :+1: